### PR TITLE
Add aws_security_group_rule_invalid_protocol rule

### PR DESCRIPTION
**Pull request by x-color**
Add `aws_security_group_rule_invalid_protocol` rule.

This rule checks whether `protocol` has an invalid value as protocol.

### CODE REVIEW

File: rules/aws_security_group_rule_invalid_protocol.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// NewAwsSecurityGroupRuleInvalidProtocolRule returns new rule with default attributes
+func NewAwsSecurityGroupRuleInvalidProtocolRule() *AwsSecurityGroupRuleInvalidProtocolRule {
+	return &AwsSecurityGroupRuleInvalidProtocolRule{
+		resourceType:  "aws_security_group_rule",
+		attributeName: "protocol",
+		protocols: map[string]struct{}{
+			"tcp":    {},
+			"udp":    {},
+			"icmp":   {},
+			"icmpv6": {},
```
Comment (wata727) [inline rules/aws_security_group_rule_invalid_protocol.go:31]: `all` is also accepted as protocol.
https://github.com/hashicorp/terraform-provider-aws/blob/724c62ba7cea20d512187d83db1ed384080a3e92/internal/service/ec2/vpc_security_group.go#L1330-L1344

File: docs/rules/aws_security_group_rule_invalid_protocol.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# aws_security_group_rule_invalid_protocol
```
Comment (wata727) [inline docs/rules/aws_security_group_rule_invalid_protocol.md:1]: Please add a link to this rule in `README.md.tmpl`. I think this rule is categorized as possible errors.
https://github.com/terraform-linters/tflint-ruleset-aws/blob/45cd608fd65235483faa57caa1da9024feb99f1e/docs/rules/README.md.tmpl

**Review by wata727 — COMMENTED** [review]
Nice! I left 2 suggestions, but I believe this rule is useful.

File: rules/aws_security_group_rule_invalid_protocol.go (hunk 1 above)
Comment (x-color) [reply rules/aws_security_group_rule_invalid_protocol.go:32]: Thank you for reviewing my PR!
I fixed it. e050a3c95f18fad648d68bb68b192c8ea0290102

**Comment by wata727** [comment]
Thanks!

**Pull request merged**